libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -212,12 +223,19 @@ def fmt_err(x: str) -> str:
         )
         index_client.create_index(index)
     # Create the search client
-    return SearchClient(
+    client = SearchClient(
+        endpoint=endpoint,
+        index_name=index_name,
+        credential=credential,
+        user_agent=user_agent,
+    )
+    async_client = AsyncSearchClient(
         endpoint=endpoint,
         index_name=index_name,
         credential=credential,
         user_agent=user_agent,
     )
+    return client, async_client
```

Comment:
is it reasonable to only instantiate the one that's needed? are the clients stateful?

Reply:
how do we know which is needed? you're saying for user to specify at vecstore instantiation if client should be sync or async? or to instantiate lazily?